src/main/java/de/maxhenkel/peek/data/DataStore.java

```diff
@@ -1,12 +1,14 @@
 package de.maxhenkel.peek.data;
 
+import net.minecraft.core.NonNullList;
+import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.block.entity.ShulkerBoxBlockEntity;
 
 import javax.annotation.Nullable;
 
 public class DataStore {
-
     @Nullable
     public static ShulkerBoxBlockEntity lastOpenedShulkerBox;
-
+    public static NonNullList<ItemStack> ecinv = NonNullList.withSize(27, ItemStack.EMPTY);
```

Comment:
This should be `null` by default to indicate it not being cached. This should also then be annotated with `@Nullable`